testutil/beaconmock/headproducer.go

```diff
@@ -176,7 +197,28 @@ func (p *headProducer) handleEvents(w http.ResponseWriter, r *http.Request) {
 	query := r.URL.Query()
 	query.Set("stream", sseStreamID) // Add sseStreamID for sse server to serve events on.
 	r.URL.RawQuery = query.Encode()
-	p.server.ServeHTTP(w, r)
+
+	for _, topic := range query["topics"] {
```

Comment:
this will break if multiple topics requested

Reply:
yeah initially i tried with single server that handles different topics but on the vouch side it has 2 different handlers for head and block events and it panics if head event handler gets block event and vice versa: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/service.go#L77

Here's the head event handler: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/events.go#L39

Here's the block event handler: https://github.com/attestantio/vouch/blob/9b8114dc0374d7e5bb6b01ce5f519213bfe30d37/services/cache/standard/events.go#L27